include/drivers/pwm.h

```diff
@@ -87,6 +116,94 @@ static inline int z_impl_pwm_pin_set_cycles(struct device *dev, uint32_t pwm,
 	return api->pin_set(dev, pwm, period, pulse, flags);
 }
 
+
+#ifdef CONFIG_PWM_CAPTURE_API
+
+/**
+ * @brief Capture the period and pulse width for a single PWM output.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pwm PWM pin.
+ * @param period Pointer to where the period (in clock cycles) is to
+ *               be stored. HW specific.
+ * @param pulse Pointer to where the pulse width (in clock cycles) is
+ *              to be stored. HW specific.
+ * @param flags Flags for pin configuration (polarity).
+ * @param timeout Non-negative waiting period for capturing
+ *                or one of the special values K_NO_WAIT and
+ *                K_FOREVER.
+ * @retval 0 If successful.
+ * @retval -EINVAL if invalid function parameters were given
+ * @retval -ENOTSUP if PWM capture is not supported
+ * @retval -EBUSY returned without waiting
+ * @retval -EAGAIN waiting period timed out
+ * @retval -EIO if an IO error occurred while capturing
+ */
+__syscall int pwm_pin_capture_cycles(struct device *dev, uint32_t pwm,
+				     uint32_t *period, uint32_t *pulse,
+				     pwm_flags_t flags, k_timeout_t timeout);
+
+static inline int z_impl_pwm_pin_capture_cycles(struct device *dev,
+						uint32_t pwm,
+						uint32_t *period,
+						uint32_t *pulse,
+						pwm_flags_t flags,
+						k_timeout_t timeout)
+{
+	struct pwm_driver_api *api;
```

Comment:
The public API also should not be excluded.

I'm not sure if it's sufficient to leave the `__syscall()` declaration exposed but exclude the `z_impl` inline, or if the `z_impl` inline should be present but just return `-ENOTSUP`.   I've generally assumed the latter, for which I like the following as an idiom, but won't insist on it.
```
static inline int z_impl_foo (args)
{
   int ret = -ENOTSUP;

#ifdef CONFIG_SELECTOR
   blah blah
   ret = api->fn(args);
#endif

   return ret;
}
```